### ISSUE TITLE
Match trailing comma in Source and Declaration Files

### ISSUE THREAD

**Issue by simonua** [Microsoft/TypeScript#17206]
**TypeScript Version:**  2.4.1

**Code**

```ts
enum Colors {
    Blue = "BLUE",
    Green = "GREEN",
    Red = "RED"
}
```

tsconfig.json
```
{
    "compilerOptions": {
        "declaration": true
    },
    "include": [
        "**/*.ts"
    ]
}
```

**Expected behavior:**
The TypeScript Declaration file should match the source file regarding trailing commas. Thus, the output should be this:
```ts
declare enum Colors {
    Blue = "BLUE",
    Green = "GREEN",
    Red = "RED"
}
```

**Actual behavior:**
Trailing comma for `Red`.
```ts
declare enum Colors {
    Blue = "BLUE",
    Green = "GREEN",
    Red = "RED",
}
```

This triggers an issue for TSLint's _[no-trailing-comma](https://palantir.github.io/tslint/rules/trailing-comma/)_ rule.

#16974

**Comment by prashaantt** [comment]
FWIW, trailing commas are gaining wider acceptance in ES2017. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas.

**Comment by SlurpTheo** [comment]
`*.d.ts` files output from 2.9.2 seem to have dropped the trailing commas... can't find mention of the issue/PR where that happened though (nor if there's some sort of flag to control this automatic change which surprised me a bit).
